sdk/eventhub/Azure.Messaging.EventHubs/src/Core/Guard.cs

```diff
@@ -3,87 +3,31 @@
-        /// <param name="argumentValue">The value of the argument to verify.</param>
-        ///
-        public static void ArgumentNotNull(string argumentName,
-                                           object argumentValue)
-        {
-            if (argumentValue == null)
-            {
-                throw new ArgumentNullException(argumentName);
-            }
-        }
-
-        /// <summary>
-        ///   Ensures that an argument's value is not <c>null</c> or an empty string, throwing an
-        ///   <see cref="ArgumentException" /> if that invariant is not met.
-        /// </summary>
-        ///
-        /// <param name="argumentName">The name of the argument being considered.</param>
-        /// <param name="argumentValue">The value of the argument to verify.</param>
-        ///
-        public static void ArgumentNotNullOrEmpty(string argumentName,
-                                                  string argumentValue)
-        {
-            if (String.IsNullOrEmpty(argumentValue))
-            {
-                throw new ArgumentException(String.Format(CultureInfo.CurrentCulture, Resources.ArgumentNullOrEmpty, argumentName), argumentName);
-            }
-        }
-
-        /// <summary>
-        ///   Ensures that an argument's value is not <c>null</c> or a string comprised of only whitespace,
-        ///   throwing an <see cref="ArgumentException" /> if that invariant is not met.
-        /// </summary>
-        ///
-        /// <param name="argumentName">The name of the argument being considered.</param>
-        /// <param name="argumentValue">The value of the argument to verify.</param>
-        ///
-        public static void ArgumentNotNullOrWhitespace(string argumentName,
-                                                       string argumentValue)
-        {
-            if (String.IsNullOrWhiteSpace(argumentValue))
-            {
-                throw new ArgumentException(String.Format(CultureInfo.CurrentCulture, Resources.ArgumentNullOrWhiteSpace, argumentName), argumentName);
-            }
-        }
-
         /// <summary>
         ///   Ensures that an argument's value is a string comprised of only whitespace, though
         ///   <c>null</c> is considered a valid value.  An <see cref="ArgumentException" /> is thrown
         ///   if that invariant is not met.
         /// </summary>
         ///
-        /// <param name="argumentName">The name of the argument being considered.</param>
         /// <param name="argumentValue">The value of the argument to verify.</param>
-        ///
-        public static void ArgumentNotEmptyOrWhitespace(string argumentName,
-                                                        string argumentValue)
+        /// <param name="argumentName">The name of the argument being considered.</param>
+        /// <exception cref="ArgumentException">The argument is empty or contains only white-space.</exception>
```

Comment:
For consistency with other types in Event Hubs, I'd prefer that we remove the `<exception>` lines, but if you feel strongly about them, please add a blank before and after to match the formatting for other doc comments.

Reply:
What would be a reason for not documenting known possible extensions, similar to what .NET and other APIs do?

Reply:
Not strongly, no, but it does help developers - especially with developer exceptions (e.g. parameter X can't be null). Using nullability attributes throughout our code would be nice, but even if/when we do if code calling into us doesn't during compilation it's still not checked, meaning we still have to do runtime validation. Methods deeper down the call stack could throw plenty of other exceptions. I guess I always saw documented exceptions in the Framework as "here's some exceptions you should expect and probably handle". Less important, I admit, for developer exceptions but the fact they are documented does help indicate that certain parameters cannot be null/empty/out of range/etc.

So I'll probably keep these since `Argument` has them but I promise I won't touch other parts of the code. ;)

Reply:
Good enough for me.  :)